File: app/Controllers/inscription.py
from app.Models.inscription import Inscription
from fastapi import HTTPException
from services.fiuba import FiubaService
from app.Models.subject import Subject


class InscriptionController():

    def post(self, reposity, body):

        student_id = body.student_id
        subject_code = body.subject_code

        fiuba_service = FiubaService()

        if fiuba_service.validateStudentId(student_id) and Subject.exists(subject_code):
            inscription = Inscription(student_id, subject_code)

            reposity.append(inscription)

            return {'message': 'inscription created'}

        raise HTTPException(status_code=400, detail='inscription error')


    def get(self, repository):
        inscriptions = [inscription.toJson() for inscription in repository]
        return {'inscriptions': inscriptions}

File: src/Controllers/inscription.js
'use strict'
const Inscription = require('../Models/inscription');
const FiubaService = require('../Services/fiuba');
var inscription_repository = []

class InscriptionController {
    async createInscription(req, res){

        const student_id = req.body.student_id;
        const subject_code = req.body.subject_code;
        
        const valid_student_id = await FiubaService.validateStudentId(student_id);
        
        if(Inscription.exists(subject_code) && valid_student_id){
            const inscription = new Inscription(student_id, subject_code);
            inscription_repository.push(inscription);

            res.status(201).json({message: 'inscription created'});
        }else{
            res.status(400).json({message: 'inscription error'});
        }
        
    }

    async getInscriptions(req, res){
        res.status(200).json({inscriptions: inscription_repository});
    }
    
    async reset(req, res){
        inscription_repository = []
        res.status(200).json({messsage: "ok"});
    }
}

module.exports = InscriptionController;